sklearn/gaussian_process/gpr.py

```diff
@@ -131,14 +137,21 @@ def optimizer(obj_func, initial_theta, bounds):
     """
     def __init__(self, kernel=None, alpha=1e-10,
                  optimizer="fmin_l_bfgs_b", n_restarts_optimizer=0,
-                 normalize_y=False, copy_X_train=True, random_state=None):
+                 normalize_y=False, copy_X_train=True, random_state=None,
+                 n_jobs=1):
         self.kernel = kernel
         self.alpha = alpha
         self.optimizer = optimizer
         self.n_restarts_optimizer = n_restarts_optimizer
         self.normalize_y = normalize_y
         self.copy_X_train = copy_X_train
         self.random_state = random_state
+        self.n_jobs = n_jobs
+
+    def _optima_iterations(self, optima, bounds, theta_initial, obj_func):
+        optima.append(
```

Comment:
How can that work ? If you read my previous https://github.com/scikit-learn/scikit-learn/pull/7997#discussion_r91288573, when n_jobs!=1, each worker gets a copy of `optima` and  this should not change `optima` in the main process ...

You should return `self._constrained_optimizations(...)` and do the `optima.append` in `fit`.